### DISCUSSION TITLE
Workspace to span multiple outputs

### DISCUSSION THREAD

**Discussion by zalegrala** [i3/i3#6167] [Q&A] [answered]
https://i3wm.org/docs/userguide.html#workspace_screen

Section 4.20 seems to indicate that a workspace can span multiple outputs.  I've acquired a new monitor, which has the behavior of the Dell described in those docs, where `xrandr` shows two outputs when I'm on the thunderbolt connection.  However, with a config of `workspace 5 output DP-1 DP-2` the workspace doesn't span both outputs.

Additionally, I thought `i3-msg workspace 5 output DP-1 DP-2` would work, but the terminal executing the command disappears until I switch away from the workspace and come back.

I did create a keybinding for `fullscreen toggle global` which does work reliably.

Is this a bug, or do I perhaps misunderstand the behavior?  I'd appreciate any pointers.

i3: v4.23

**Comment by okraits** [comment, marked as answer]
1. Section 4.20 says: `You can specify multiple outputs. The first available will be used.`
2. `i3-msg` can only be used to execute commands. The `workspace` directive explained in section 4.20 is not a command.
3. You can combine several monitors to a virtual screen: https://wiki.archlinux.org/title/Multihead#Combine_screens_into_virtual_display Then i3 will put a workspace on that virtual screen.

**Reply by zalegrala** [reply]
Thank you for the link, I think that's what I was missing.  I've now combined the outputs into a single virtual screen and it is working as expected.
